app/partials/app.cjsx

```diff
@@ -12,35 +13,53 @@ PanoptesApp = React.createClass
     initialLoadComplete: React.PropTypes.bool
     user: React.PropTypes.object
     geordi: React.PropTypes.object
+    notificationsCounter: React.PropTypes.object
+    unreadNotificationsCount: React.PropTypes.number
 
   getChildContext: ->
     initialLoadComplete: @state.initialLoadComplete
     user: @state.user
     geordi: @geordiLogger
+    notificationsCounter: @state.notificationsCounter
+    unreadNotificationsCount: @state.unreadNotificationsCount
 
   getInitialState: ->
     initialLoadComplete: false
     user: null
+    notificationsCounter: new NotificationsCounter()
 
   componentWillMount: ->
     @geordiLogger = new GeordiLogger
-  
+
   componentDidMount: ->
+    @state.notificationsCounter.listen (unreadNotificationsCount) =>
+      @setState {unreadNotificationsCount}
+
     auth.listen 'change', @handleAuthChange
     generateSessionID()
     @handleAuthChange()
 
   componentWillUnmount: ->
     auth.stopListening 'change', @handleAuthChange
 
+  componentWillReceiveProps: (nextProps) ->
+    @updateNotificationsCount params: nextProps.params
```

Comment:
Right, that's the only props, but I guess I'm just thinking about if in the future there are any other props, we might want an explicit check, but that doesn't need to happen now.